src/plugins/intel_cpu/src/emitters/snippets/x64/jit_loop_emitters.cpp

```diff
@@ -14,6 +14,52 @@ using namespace dnnl::impl::cpu::x64;
 namespace ov {
 namespace intel_cpu {
 
+namespace {
+class jit_aux_gpr_holder {
+public:
+    jit_aux_gpr_holder(dnnl::impl::cpu::x64::jit_generator* host, std::vector<size_t>& pool_gpr_idxs, const std::vector<size_t>& used_gpr_idxs)
+        : m_h(host), m_pool_gpr_idxs(pool_gpr_idxs) {
+        // If the pool is empty, let's manualy allocate the gpr and push original vlaue on stack
+        if (m_pool_gpr_idxs.empty()) {
+            m_aux_gpr_idx = Reg64(static_cast<int>(allocate_aux_gpr(used_gpr_idxs)));
+            m_is_preserved = true;
+            m_h->push(m_aux_gpr_idx);
+        } else {
+            m_aux_gpr_idx = Reg64(static_cast<int>(m_pool_gpr_idxs.back()));
+            m_pool_gpr_idxs.pop_back();
+        }
+    }
+
+    ~jit_aux_gpr_holder() {
+        if (m_is_preserved) {
+            m_h->pop(m_aux_gpr_idx);
+        } else {
+            m_pool_gpr_idxs.push_back(m_aux_gpr_idx.getIdx());
+        }
+    }
+
+    const Reg64& get_reg() const { return m_aux_gpr_idx; }
+
+private:
+    size_t allocate_aux_gpr(const std::vector<size_t>& used_gpr_idxs) const {
+        // RSP, RBP - stack-related registers, abi_param1 - runtime parameter register in the kernel
+        static std::set<size_t> blakclist_gpr_idxs = { Operand::RSP, Operand::RBP, static_cast<size_t>(abi_param1.getIdx()) };
+        for (size_t gpr_idx = 0; gpr_idx <= Operand::R15; ++gpr_idx) {
+            size_t _idx = Operand::R15 - gpr_idx; // we allocate from the end
+            if (std::find(used_gpr_idxs.cbegin(), used_gpr_idxs.cend(), _idx) != used_gpr_idxs.cend()) continue;
+            if (std::find(blakclist_gpr_idxs.cbegin(), blakclist_gpr_idxs.cend(), _idx) != blakclist_gpr_idxs.cend()) continue;
```

Comment:
Why don't we use `count(_idx)`? It has log complexity instead of linear.

Reply:
I'd prefer to align with `used_gpr_idxs` when we use `std::find`. In my opinion, the code:
```
if (std::find(used_gpr_idxs.cbegin(), used_gpr_idxs.cend(), _idx) != used_gpr_idxs.cend()) continue;
if (std::find(blakclist_gpr_idxs.cbegin(), blakclist_gpr_idxs.cend(), _idx) != blakclist_gpr_idxs.cend()) continue;
```
 is prettier than:
 ```
if (std::find(used_gpr_idxs.cbegin(), used_gpr_idxs.cend(), _idx) != used_gpr_idxs.cend()) continue;
if (blakclist_gpr_idxs.count(_idx) > 0) continue;
```
😃 

Moreover `blakclist_gpr_idxs` contains just 3 elements, using `.count()` is just a drop in the ocean 😄 

Thus, I chose prettier code (for me at least) :blush:

Reply:
I don't know what to answer for that 🙃 
Should we enable benchmarks on code beauty instead of throughput and latency?
Customer: you code works 30% slower than in the previous release!
We: yes, but it looks prettier

Reply:
Agreed to leave as is due to minimal performance impact